src/encounter.c

```diff
@@ -54,982 +55,951 @@
 #include "unk_02096420.h"
 #include "vars_flags.h"
 
-typedef struct {
-    int *unk_00;
-    int unk_04;
-    int unk_08;
+typedef struct Encounter {
+    int *resultMaskPtr;
+    int introEffectID;
+    int battleBGM;
     int unk_0C;
-    FieldBattleDTO *unk_10;
-} UnkStruct_02050ACC;
-
-typedef struct {
-    int unk_00;
-    int unk_04;
-    int unk_08;
-    int *unk_0C;
-    FieldBattleDTO *unk_10;
-} UnkStruct_02050DD4;
-
-static void sub_020518B0(FieldSystem *fieldSystem, FieldBattleDTO *param1);
-static void sub_02051988(FieldSystem *fieldSystem, FieldBattleDTO *param1);
-static BOOL sub_02050EE0(FieldTask *taskMan);
-static BOOL sub_02051074(FieldTask *taskMan);
-
-static BOOL sub_02050A74(FieldTask *taskMan)
-{
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(taskMan);
-    FieldBattleDTO *v1 = FieldTask_GetEnv(taskMan);
-    int *v2 = FieldTask_GetState(taskMan);
-
-    switch (*v2) {
+    FieldBattleDTO *dto;
+} Encounter;
+
+typedef struct WildEncounter {
+    int state;
+    int introEffectID;
+    int battleBGM;
+    int *resultMaskPtr;
+    FieldBattleDTO *dto;
+} WildEncounter;
+
+static Encounter *NewEncounter(FieldBattleDTO *dto, int introEffectID, int battleBGM, int *resultMaskPtr);
+static WildEncounter *NewWildEncounter(FieldBattleDTO *dto, int introEffectID, int battleBGM, int *resultMaskPtr);
+static void FreeEncounter(Encounter *encounter);
+static void FreeWildEncounter(WildEncounter *encounter);
```

Comment:
**question:** Should these still follow the `Module_Function()` format even though they're static, for clarity/uniformity?

Reply:
We don't really have a rule for that for `static` functions; I find that the simpler format here is a bit more readable, and clearly communicates that these are `static`.